### DISCUSSION TITLE
DKI Processing

### DISCUSSION THREAD

**Discussion by mmsarhan** [muscbridge/PyDKE#275] [Technical Support]
I just ran the command to process my DKI data and I was not sure if everything was processed as I received many error messages. I have pasted the full error list that came up on my terminal. Thank you 
I just finished processing my files and this is what came up. I am not sure if everything ran correctly and wanted to ask. I have the full error messages that came up below. Thanks again for your help.

(DKE) [blab@localhost Scans]$ pydesigner -s DKI_BIPOLAR_2.5mm_64dir_50slices_20/
mrinfo: [done] scanning DICOM folder "/home/blab...LAR_2.5mm_64dir_50slices_20"
mrinfo: [100%] reading DICOM series "DKI_BIPOLAR_2.5mm_64dir_50slices"
mrconvert -quiet /home/blab/Downloads/Scans/dwi0.mif /home/blab/Downloads/Scans/working.mif
Insufficient information in .json file to determine Fourier status. Assuming DWI is partial Fourier
[WARNING] Given DWI is partial fourier, overriding --degibbs; no unringing correction will be done to avoid artifacts.Use the "--adv" flag to run forced corrections.
dwifslpreproc: [WARNING] DWI header indicates no phase encoding contrast between b=0 images; proceeding without inhomogeneity field estimation

dwifslpreproc: [ERROR] eddy* --imain=eddy_in.nii --mask=eddy_mask.nii --acqp=eddy_config.txt --index=eddy_indices.txt --bvecs=bvecs --bvals=bvals --repol --data_is_shelled --out=dwi_post_eddy --verbose (app.py:196)
dwifslpreproc: [ERROR] Information from failed command:
dwifslpreproc:
               =============
               eddy_cuda10.2
               =============
               
               
               ===========
               eddy_openmp
               ===========
               
               
               =============
               eddy_cuda10.2
               =============
               eddy_cuda10.2: error while loading shared libraries: libcublas.so.10: cannot open shared object file: No such file or directory
               
               
               ===========
               eddy_openmp
               ===========
               eddy_openmp: error while loading shared libraries: libquadmath.so.0: cannot open shared object file: No such file or directory
               
               
dwifslpreproc:
dwifslpreproc: [ERROR] For debugging, inspect contents of scratch directory: /home/blab/Downloads/Scans/dwifslpreproc-tmp-L7K02X/
Traceback (most recent call last):
  File "/home/blab/anaconda3/envs/DKE/bin/pydesigner", line 8, in <module>
    sys.exit(main())
  File "/home/blab/anaconda3/envs/DKE/lib/python3.7/site-packages/designer/pydesigner.py", line 644, in main
    verbose=args.verbose)
  File "/home/blab/anaconda3/envs/DKE/lib/python3.7/site-packages/designer/preprocessing/mrpreproc.py", line 481, in undistort
    raise Exception('dwifslpreproc failed, please look above for '
Exception: dwifslpreproc failed, please look above for error sources.
(DKE) [blab@localhost Scans]$

**Comment by TheJaeger** [comment]
FSL's eddy only works with CUDA 10.0 if you have an Nvidia GPU. You need to get uninstall CUDA 10.2 and install CUDA 10.0 in its place. if you don't have a CUDA-enabled GPU, then resolve missing `libquadmath.so.0` library by installing `gcc`.

**Reply by mmsarhan** [reply]
How would I figure out the version of CUDA I have and to unistall it and download the correct version?

**Reply by TheJaeger** [reply]
You can check your current Nvidia driver and CUDA versions with
```
nvidia-smi
nvcc -V
```
Refer to instructions [here](https://gist.github.com/Mahedi-61/e0625c8782426168e436fb98417ef209) or [here](https://linuxconfig.org/how-to-install-nvidia-cuda-toolkit-on-centos-7-linux) on how to install specific CUDA version on CentOS. There are countless other instructions you may refer to across the web.

Just be sure that you have a CUDA-enabled GPU physically present in your computer before you begin troubleshooting.

**Reply by mmsarhan** [reply]
I figured out that i do not have CUDA on my computer and downloaded GCC. I checked the version and it say I downloaded version 4.8.5. Is there anything else I need to install so that the DKI processing will work?

**Reply by TheJaeger** [reply]
Give it a shot again to see if it resolved the error.

**Reply by TheJaeger** [reply]
If it's still a problem, I suggest running the [NeuroDock](https://hub.docker.com/r/dmri/neurodock) container. It should theoretically give you better performance than a VM and is guaranteed to run flawlessly.

**Reply by mmsarhan** [reply]
@TheJaeger 
I tried running neurodock, but I am not sure the installation went correctly. I was able to download the Docker for Windows application, but when I went to go to the terminal and try to download neurodocks and received this error message:
error response from daemon manifest for dmri/neurodock:v0.2 not found: manifest unknown: manifest unknown
